### ISSUE TITLE
--get-thumbnail option gives url to "broken" images.

### ISSUE THREAD

**Issue by mhogomchungu** [yt-dlp/yt-dlp#402]
## Checklist

- [x ] I'm reporting a broken site support issue
- [x ] I've verified that I'm running yt-dlp version **2021.06.09**
- [x ] I've checked that all provided URLs are alive and playable in a browser
- [x ] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x ] I've searched the bugtracker for similar bug reports including closed ones
- [x ] I've read bugs section in FAQ

## Verbose log

```
youtube-dl output

youtube-dl --get-thumbnail https://www.youtube.com/channel/UCiIws_onINIcqZ3SUjSbnVg

https://i.ytimg.com/vi/Mepj-RQaCF4/hqdefault.jpg?sqp=-oaymwEcCNACELwBSFXyq4qpAw4IARUAAIhCGAFwAcABBg==&rs=AOn4CLDFhKY2Kq-AZISt6eODLOWdkStC6g
https://i.ytimg.com/vi/RvBqNg4Pppw/hqdefault.jpg?sqp=-oaymwEcCNACELwBSFXyq4qpAw4IARUAAIhCGAFwAcABBg==&rs=AOn4CLAaZjG7k0ONnEnNAdUd1roe3W2h9g
https://i.ytimg.com/vi/OU7lWJBZA7A/hqdefault.jpg?sqp=-oaymwEcCNACELwBSFXyq4qpAw4IARUAAIhCGAFwAcABBg==&rs=AOn4CLB3GJA0Xl8QgPwZooPz_bHk_zqYOg
https://i.ytimg.com/vi/GuP4h_6CsAY/hqdefault.jpg?sqp=-oaymwEcCNACELwBSFXyq4qpAw4IARUAAIhCGAFwAcABBg==&rs=AOn4CLBp-bKyzC6zh4A-feJoiqb4VPzmiA

```
```
yt-dlp output:

yt-dlp --get-thumbnail https://www.youtube.com/channel/UCiIws_onINIcqZ3SUjSbnVg 
WARNING: [youtube:tab] A channel/user page was given. All the channel's videos will be downloaded. To download only the videos in the home page, add a "/featured" to the URL
https://i.ytimg.com/vi/ObfrixWOOQQ/maxresdefault.jpg
https://i.ytimg.com/vi/B0aBRbCOzK8/maxresdefault.jpg
https://i.ytimg.com/vi/v5vnApl0S44/maxresdefault.jpg
https://i.ytimg.com/vi/sSwtoGg4zK4/maxresdefault.jpg
```

## Description

The "--get-thumbnail" option works as expected in youtube-dl because it gives thumbnails that are the same as seen in the browser and yt-dl gives different links that show a generic "blank" thumbnail.

**Comment by pukkandan** [comment]
It's a side-effect of ff2751ac9cc7d4150797d3207da9b566396bc796. I'll need to figure out how to fix both this issue and #337 at the same time

**Comment by pukkandan** [comment]
Now you can use `--check-formats` to verify that the thumbnail actually exists. It works by actually requesting the thumbnail and checking for network errors. I'll see if there is some other way to handle this when I work on #340

```
$ yt-dlp https://www.youtube.com/watch?v=pS4ca4GB-_c -O thumbnail
https://i.ytimg.com/vi/pS4ca4GB-_c/maxresdefault.jpg

$ yt-dlp https://www.youtube.com/watch?v=pS4ca4GB-_c --list-thumbnail
[youtube] pS4ca4GB-_c: Downloading webpage
[info] Thumbnails for pS4ca4GB-_c:
ID width   height  URL
0  168     94      https://i.ytimg.com/vi/pS4ca4GB-_c/hqdefault.jpg?sqp=-oaymwEbCKgBEF5IVfKriqkDDggBFQAAiEIYAXABwAEG&rs=AOn4CLCBQ4_ln39dFCbgt268t8lMJUtfdA
1  196     110     https://i.ytimg.com/vi/pS4ca4GB-_c/hqdefault.jpg?sqp=-oaymwEbCMQBEG5IVfKriqkDDggBFQAAiEIYAXABwAEG&rs=AOn4CLACk7_pc2_UMDQ3R-p4Epk4q517Jw
2  246     138     https://i.ytimg.com/vi/pS4ca4GB-_c/hqdefault.jpg?sqp=-oaymwEcCPYBEIoBSFXyq4qpAw4IARUAAIhCGAFwAcABBg==&rs=AOn4CLCoeij_0sT78RWLVeAwZ_oQZhb8RQ
3  336     188     https://i.ytimg.com/vi/pS4ca4GB-_c/hqdefault.jpg?sqp=-oaymwEcCNACELwBSFXyq4qpAw4IARUAAIhCGAFwAcABBg==&rs=AOn4CLCC9H7tB1SUKxNjuEFpeLGHh4vZxQ
4  480     360     https://i.ytimg.com/vi/pS4ca4GB-_c/hqdefault.jpg
5  unknown unknown https://i.ytimg.com/vi/pS4ca4GB-_c/maxresdefault.jpg
```

```
$ yt-dlp https://www.youtube.com/watch?v=pS4ca4GB-_c -O thumbnail --check-formats
https://i.ytimg.com/vi/pS4ca4GB-_c/hqdefault.jpg

$ yt-dlp https://www.youtube.com/watch?v=pS4ca4GB-_c --list-thumbnail --check-formats
[youtube] pS4ca4GB-_c: Downloading webpage
[info] Testing thumbnail 5
[info] Unable to connect to thumbnail 5 URL "https://i.ytimg.com/vi/pS4ca4GB-_c/maxresdefault.jpg" - HTTP Error 404: Not Found. Skipping...
[info] Testing thumbnail 4
[info] Testing thumbnail 3
[info] Testing thumbnail 2
[info] Testing thumbnail 1
[info] Testing thumbnail 0
[info] Thumbnails for pS4ca4GB-_c:
ID width height URL
0  168   94     https://i.ytimg.com/vi/pS4ca4GB-_c/hqdefault.jpg?sqp=-oaymwEbCKgBEF5IVfKriqkDDggBFQAAiEIYAXABwAEG&rs=AOn4CLCBQ4_ln39dFCbgt268t8lMJUtfdA
1  196   110    https://i.ytimg.com/vi/pS4ca4GB-_c/hqdefault.jpg?sqp=-oaymwEbCMQBEG5IVfKriqkDDggBFQAAiEIYAXABwAEG&rs=AOn4CLACk7_pc2_UMDQ3R-p4Epk4q517Jw
2  246   138    https://i.ytimg.com/vi/pS4ca4GB-_c/hqdefault.jpg?sqp=-oaymwEcCPYBEIoBSFXyq4qpAw4IARUAAIhCGAFwAcABBg==&rs=AOn4CLCoeij_0sT78RWLVeAwZ_oQZhb8RQ
3  336   188    https://i.ytimg.com/vi/pS4ca4GB-_c/hqdefault.jpg?sqp=-oaymwEcCNACELwBSFXyq4qpAw4IARUAAIhCGAFwAcABBg==&rs=AOn4CLCC9H7tB1SUKxNjuEFpeLGHh4vZxQ
4  480   360    https://i.ytimg.com/vi/pS4ca4GB-_c/hqdefault.jpg
```
